discounts/rust/order-discounts/fixed-amount/src/main.rs

```diff
@@ -0,0 +1,159 @@
+use serde::{Deserialize, Serialize};
+
+mod api;
+use api::*;
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Configuration {
+    pub value: f64,
+}
+
+impl Configuration {
+    const DEFAULT_VALUE: f64 = 50.00;
+
+    fn from_str(str: &str) -> Self {
+        serde_json::from_str(str).expect("Unable to parse configuration value from metafield")
+    }
+}
+
+impl Default for Configuration {
+    fn default() -> Self {
+        Configuration {
+            value: Self::DEFAULT_VALUE,
+        }
+    }
+}
+
+impl input::Input {
+    pub fn configuration(&self) -> Configuration {
+        match &self.discount_node.metafield {
+            Some(input::Metafield { value }) => Configuration::from_str(value),
+            None => Configuration::default(),
+        }
+    }
+}
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let input: input::Input = serde_json::from_reader(std::io::BufReader::new(std::io::stdin()))?;
+    let mut out = std::io::stdout();
+    let mut serializer = serde_json::Serializer::new(&mut out);
+    function(input)?.serialize(&mut serializer)?;
+    Ok(())
+}
+
+fn function(input: input::Input) -> Result<FunctionResult, Box<dyn std::error::Error>> {
+    let config = input.configuration();
+    let converted_value = convert_to_cart_currency(config.value, input.presentment_currency_rate);
+    let targets = vec![Target::OrderSubtotal {
+        excluded_variant_ids: vec![],
+    }];
+    Ok(build_result(converted_value, targets))
+}
+
+fn convert_to_cart_currency(value: f64, presentment_currency_rate: f64) -> f64 {
+    value * presentment_currency_rate
+}
+
+fn build_result(amount: f64, targets: Vec<Target>) -> FunctionResult {
+    let discounts = if targets.is_empty() {
+        vec![]
+    } else {
+        vec![Discount {
+            message: None,
+            conditions: None,
+            targets,
+            value: Value::FixedAmount { amount },
+        }]
+    };
+    FunctionResult {
+        discounts,
+        discount_application_strategy: DiscountApplicationStrategy::First,
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    impl Default for input::Input {
+        fn default() -> Self {
+            let json = r#"
+            {
+                "discountNode": { "metafield": null },
+                "presentmentCurrencyRate": "1.00"
```

Comment:
@adampetro @DuncanUszkay1 Is this a reasonable test setup for you? I think this way we can catch failures like converting this value from String, and still override with structs in the test